app/src/main/java/org/gdg/frisbee/android/utils/EndlessScrollListener.java

```diff
@@ -0,0 +1,70 @@
+package org.gdg.frisbee.android.utils;
+
+import android.widget.AbsListView;
+
+public abstract class EndlessScrollListener implements AbsListView.OnScrollListener {
+    // The minimum number of items to have below your current scroll position
+    // before loading more.
+    private int mVisibleThreshold = 5;
+    // The current offset index of data you have loaded
+    private int mCurrentPage = 0;
+    // The total number of items in the dataset after the last load
+    private int mPreviousTotalItemCount = 0;
+    // True if we are still waiting for the last set of data to load.
+    private boolean mLoading = true;
+    // Sets the starting page index
+    private int mStartingPageIndex = 0;
+
+    public EndlessScrollListener() {
+    }
+
+    public EndlessScrollListener(int visibleThreshold) {
+        mVisibleThreshold = visibleThreshold;
+    }
+
+    public EndlessScrollListener(int visibleThreshold, int startPage) {
+        mVisibleThreshold = visibleThreshold;
+        mStartingPageIndex = startPage;
+        mCurrentPage = startPage;
+    }
+
+    // This happens many times a second during a scroll, so be wary of the code you place here.
+    // We are given a few useful parameters to help us work out if we need to load some more data,
+    // but first we check if we are waiting for the previous load to finish.
+    @Override
+    public void onScroll(AbsListView view, int firstVisibleItem, int visibleItemCount, int totalItemCount) {
+        // If the total item count is zero and the previous isn't, assume the
+        // list is invalidated and should be reset back to initial state
+        if (totalItemCount < mPreviousTotalItemCount) {
+            mCurrentPage = mStartingPageIndex;
+            mPreviousTotalItemCount = totalItemCount;
+            if (totalItemCount == 0) {
+                mLoading = true;
```

Comment:
No, this just assumes that if the totalItems in the List is 0 we're loading the items somewhere else. It won't call the onLoadMore callback